src/languageservice/utils/schemaUrls.ts

```diff
@@ -22,3 +24,48 @@ export function checkSchemaURI(
     return uri;
   }
 }
+
+/**
+ * Collect all urls of sub schemas
+ * @param schema the root schema
+ * @returns map url to schema
+ */
+export function getSchemaUrls(schema: JSONSchema): Map<string, JSONSchema> {
+  const result = new Map<string, JSONSchema>();
+  if (!schema) {
+    return result;
+  }
+  const url = schema.url;
```

Comment:
Init the constant inside the following `if` statement or not create a constant at all.